NEXTCLOUD.cmake

```diff
@@ -5,7 +5,14 @@ set( APPLICATION_DOMAIN     "nextcloud.com" )
 set( APPLICATION_VENDOR     "Nextcloud GmbH" )
 set( APPLICATION_UPDATE_URL "https://updates.nextcloud.org/client/" CACHE STRING "URL for updater" )
 set( APPLICATION_HELP_URL   "" CACHE STRING "URL for the help menu" )
-set( APPLICATION_ICON_NAME  "Nextcloud" )
+
+if(APPLE AND APPLICATION_NAME STREQUAL "Nextcloud")
+    set( APPLICATION_ICON_NAME "Nextcloud-macOS" )
+    message("Using macOS-specific application icon: ${APPLICATION_ICON_NAME}")
+else()
```

Comment:
but should we not also offer this better macOS icon feature when `APPLICATION_NAME` is not `Nextcloud` ?

Reply:
for the other question related to the check on `APPLICATION_NAME` @claudio would it be possible to check for the existence of the icon file ?

Reply:
Yeah, added a check for the existence of the icon name

Reply:
You can find the updated version of the icon at #4714.